core/src/main/java/com/sequenceiq/cloudbreak/service/smartsense/SmartSenseSubscriptionService.java

```diff
@@ -104,27 +108,55 @@ public SmartSenseSubscription findOneById(Long id) {
         return subscriptions.hasNext() ? Optional.of(subscriptions.next()) : Optional.empty();
     }
 
-    @PostAuthorize("hasPermission(returnObject,'read')")
-    public SmartSenseSubscription getDefaultForUser(IdentityUser cbUser) {
-        SmartSenseSubscription subscription;
-        subscription = repository.findByAccountAndOwner(cbUser.getAccount(), cbUser.getUserId());
+    public SmartSenseSubscription getDefaultForUser(@Nonnull IdentityUser cbUser) {
+        requireNonNull(cbUser);
+        SmartSenseSubscription subscription = repository.findByAccountAndOwner(cbUser.getAccount(), cbUser.getUserId());
+        upgradeDefaultSmartSenseSubscription(subscription);
+        subscription = Optional.ofNullable(subscription).orElseGet(() -> createSubscriptionFromIdentityUser(cbUser));
+        checkForSmartSenseSubscriptionExistence(subscription);
+        checkSmartSenseSubscriptionAuthorization(subscription);
+        return subscription;
+    }
+
+    private void upgradeDefaultSmartSenseSubscription(SmartSenseSubscription subscription) {
         if (subscription != null && !StringUtils.isEmpty(defaultSmartsenseId) && !defaultSmartsenseId.equals(subscription.getSubscriptionId())) {
             LOGGER.info("Upgrading default SmartSense subscription");
             subscription.setSubscriptionId(defaultSmartsenseId);
             repository.save(subscription);
         }
-        return Optional.ofNullable(subscription).orElseGet(() -> {
-            SmartSenseSubscription newSubscription = null;
-            if (!StringUtils.isEmpty(defaultSmartsenseId)) {
-                LOGGER.info("Generating default SmartSense subscription");
-                newSubscription = new SmartSenseSubscription();
-                newSubscription.setSubscriptionId(defaultSmartsenseId);
-                newSubscription.setAccount(cbUser.getAccount());
-                newSubscription.setOwner(cbUser.getUserId());
-                newSubscription.setPublicInAccount(true);
-                repository.save(newSubscription);
-            }
-            return newSubscription;
-        });
     }
+
+    private void checkForSmartSenseSubscriptionExistence(SmartSenseSubscription subscription) {
+        if (subscription == null) {
+            String message = "Unable to find SmartSense subscription for the user.";
+            LOGGER.info(message);
+            throw new SmartSenseNotFoundException(message);
```

Comment:
According to security reasons [related ticket](https://hortonworks.jira.com/browse/BUG-92881) we should return with the same status code for requests when an entity doesn't exist or the user doesn't have enough permissions. Conventionally we should use the Forbidden status code, so to follow the convention we should use the AccessDenied related exception in this case too and the very same message should be used both of the edge cases.
I would merge the two method into one for checking non-existence and authority.